pkg/certifier/scorecard/scorecard_test.go

```diff
@@ -0,0 +1,249 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	mocks "github.com/guacsec/guac/internal/testing/mock"
+	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/assembler/graphdb"
+	"github.com/guacsec/guac/pkg/certifier"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+type mockScorecard struct{}
+
+func (m mockScorecard) GetScore(repoName, commitSHA string) (*pkg.ScorecardResult, error) {
+	return &pkg.ScorecardResult{}, nil
+}
+
+func TestNewScorecard(t *testing.T) {
+	tests := []struct {
+		name          string
+		sc            Scorecard
+		client        graphdb.Client
+		want          certifier.Certifier
+		wantErr       bool
+		authToken     string
+		wantAuthToken bool
+	}{
+		{
+			name:    "scorecard is nil",
+			wantErr: true,
+		},
+		{
+			name:          "Auth token is set",
+			sc:            mockScorecard{},
+			want:          &scorecard{scorecard: mockScorecard{}, ghToken: "test"},
+			authToken:     "test",
+			wantAuthToken: true,
+		},
+		{
+			name:          "Auth token is empty",
+			sc:            mockScorecard{},
+			authToken:     "",
+			wantAuthToken: true,
+			wantErr:       true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.wantAuthToken {
+				t.Setenv("GITHUB_AUTH_TOKEN", test.authToken)
+			}
+
+			got, err := NewScorecardCertifier(test.sc, test.client)
+			if (err != nil) != test.wantErr {
+				t.Errorf("NewScorecardCertifier() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("NewScorecardCertifier() got = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func Test_CertifyComponent(t *testing.T) {
+	ctx, _ := context.WithTimeout(context.Background(), time.Second) // nolint:govet
```

Comment:
add `defer cancel()`, also think we could give a bit more time maybe 2 or 3 seconds for system variance

Reply:
👍 Added!